### PR TITLE
Add/catch duplicated comment error

### PR DESCRIPTION
**Pull request by lupus2k**
Fixes https://github.com/Automattic/dotcom-forge/issues/5074

## Proposed changes:
We catch the duplicated comment error message and show it with a custom design (like we are doing when submitting a comment)
This also fixes that in certain cases, the back button is not returning to the comment form.

| Before | After |
| --- | --- |
| https://github.com/Automattic/jetpack/assets/402286/e8d46519-61e6-481d-bc33-5f1c570f096e | https://github.com/Automattic/jetpack/assets/402286/808f4f14-f75a-46dc-ad4c-28926a175b9f |

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* You need to enable the `Let visitors use a WordPress.com or Facebook account to comment.` option on `wordpress.com/settings/discussion/:your-site`
* Go to a post
* Comment something
* After reloading, comment the same comment as before
* You should receive a Duplicated comment error with better look.
* The back button should return you to the comment form.
* Test it in Chrome, Firefox, Safari, Brave...

### CODE REVIEW

**Comment by crisbusquets** [comment]
Hi @lupus2k, thanks for the ping!

How can I easily test this? I tried with a site, but I see the old implementation:

Screenshot 2024-01-08 at 13 02 21

**Comment by lupus2k** [comment]
> Hi @lupus2k, thanks for the ping!
> 
> How can I easily test this? I tried with a site, but I see the old implementation:
> 

Hey @crisbusquets, I updated the test site 1 and now it works: https://explorersatomicsub1.wpcomstaging.com/2023/12/26/boost-your-productivity-with-this-keyboard-shortcut-hack

File: projects/plugins/jetpack/modules/comments/comments.php [hunk 1]
```diff
@@ -673,6 +674,82 @@ public function subscription_modal_status_track_event() {
...
+					text-decoration: none;
+					color: #333 !important;
+				}
+			</style>
+		</head>
+		<body>
+		<div>
+			<h1>
+				<?php
+					echo esc_html__( 'Duplicate comment detected; it looks as though you&#8217;ve already said that!', 'jetpack' );
```
Comment (jeherve) [inline projects/plugins/jetpack/modules/comments/comments.php]:
```suggestion
				esc_html_e( 'Duplicate comment detected; it looks as though you’ve already said that!', 'jetpack' );
```

File: projects/plugins/jetpack/modules/comments/comments.php [hunk 2]
```diff
@@ -673,6 +674,82 @@ public function subscription_modal_status_track_event() {
...
+				?>
+			</h1>
+			<a href="javascript:history.go(-2)"><?php echo esc_html__( '&laquo; Back', 'jetpack' ); ?></a>
```
Comment (jeherve) [inline projects/plugins/jetpack/modules/comments/comments.php]:
```suggestion
			<a href="javascript:history.go(-2)"><?php esc_html_e( '&laquo; Back', 'jetpack' ); ?></a>
```

**Review by jeherve — CHANGES_REQUESTED** [review]

File: projects/plugins/jetpack/modules/comments/comments.php (hunk 2 above)
Comment (lupus2k) [reply projects/plugins/jetpack/modules/comments/comments.php]: Thanks for your suggestions @jeherve

**Comment by crisbusquets** [comment]
> Hey @crisbusquets, I updated the test site 1 and now it works: https://explorersatomicsub1.wpcomstaging.com/2023/12/26/boost-your-productivity-with-this-keyboard-shortcut-hack

Thank you @lupus!
Is the `Comment sent` text new too?

We should use a smaller font size in both cases, as it doesn't match the UI.
Can we try with the theme's h3 but in regular, and using a color that matches the theme?

**Comment by Lupus** [comment]
You probably meant to mention @lupus2k, @crisbusquets :)

**Comment by crisbusquets** [comment]
> You probably meant to mention @lupus2k, @crisbusquets :)

Ouch, yes. Sorry @Lupus 😅

**Review by jeherve — DISMISSED** [review]
The display looks good, but the back button doesn't seem to work properly. See this screencast for an example:


https://github.com/Automattic/jetpack/assets/426388/443b8ab7-7fa0-4e66-963c-4f5082095006

**Comment by lupus2k** [comment]
> Is the `Comment sent` text new too?
> 
> We should use a smaller font size in both cases, as it doesn't match the UI. Can we try with the theme's h3 but in regular, and using a color that matches the theme?

The comment sent is new, and only shows when the Subscription modal shows, but in this place, we have a `Submitting comment` message with the same style.
Should we change the styles for all of it? (it's using h1)

**Comment by lupus2k** [comment]
> The display looks good, but the back button doesn't seem to work properly. See this screencast for an example:

Interesting, I tried to reproduce without error on Chrome and Firefox, which browser are you using?

**Comment by jeherve** [comment]
This was in Firefox.

**Comment by lupus2k** [comment]
> This was in Firefox.

I was able to reproduce it, so I updated the PR and now I'm unable to do it.
Can you check using our test site: https://explorersatomicsub1.wpcomstaging.com/2023/12/26/boost-your-productivity-with-this-keyboard-shortcut-hack/

**Comment by crisbusquets** [comment]
> > This was in Firefox.
> 
> I was able to reproduce it, so I updated the PR and now I'm unable to do it. Can you check using our test site: https://explorersatomicsub1.wpcomstaging.com/2023/12/26/boost-your-productivity-with-this-keyboard-shortcut-hack/

It works for me now too

**Comment by crisbusquets** [comment]
> The comment sent is new, and only shows when the Subscription modal shows, but in this place, we have a `Submitting comment` message with the same style.
> Should we change the styles for all of it? (it's using h1)

Yes, the same styles for all of it. I think this will make it look better, h1 is way too big for something that should be a relatively small alert-notice.

**Comment by lupus2k** [comment]
> @jeherve: It works for me on that test site of yours, but I still experience issues with a fresh Jurassic Ninja site, unfortunately.

I'll check with a Jurassic Ninja site + Jetpack Beta plugin?

**Comment by jeherve** [comment]
> I'll check with a Jurassic Ninja site + Jetpack Beta plugin?

Yup. That's probably the easiest way to test PRs in this repo, with the Jetpack Live Branches tampermonkey script:
https://github.com/Automattic/jetpack/tree/trunk/tools/jetpack-live-branches

**Comment by lupus2k** [comment]
Hey @jeherve, I updated the PR, and it should work as expected on Jetpack sites & Atomic sites. I tested on Chrome, Safari, Firefox, Opera, and Brave

**Review by jeherve — DISMISSED** [review]
It tests well for me now!

**Comment by lupus2k** [comment]
Hey @crisbusquets, I just updated the PR using `h3`; here is a video with the three messages you can see:

https://github.com/Automattic/jetpack/assets/402286/94150861-4439-45d3-8a9c-7266f79c29a3

Let me know if this looks good to you; you can test it on the following sites with different themes:
- https://explorersatomicsub1.wpcomstaging.com/2023/12/26/boost-your-productivity-with-this-keyboard-shortcut-hack/
- https://explorersatomicsub2.wpcomstaging.com/2023/12/26/unveiling-the-future-a-comprehensive-review-of-the-quantumx-pro-vr-headset/
- https://explorersatomicsub3.wpcomstaging.com/2023/12/26/paradise-found-a-blissful-escape-to-emerald-isle/
- https://explorersatomicsub4.wpcomstaging.com/2023/12/26/5-minute-energizer-quick-workouts-for-busy-days/

**Comment by crisbusquets** [comment]
> Hey @crisbusquets, I just updated the PR using `h3`; here is a video with the three messages you can see:

Hi @lupus2k !

This looks better, but we should also use the theme font-family and color. If we don't, the interface looks weird. For example, in the second screenshot, both `Deja un comentario` and `Duplicate comment detected; it looks as though you’ve already said that!` are h3, but they don't look the same:

Screenshot 2024-01-11 at 11 29 18

Screenshot 2024-01-11 at 11 29 56

Something like this would look better (I only updated the font-family and font-size, so they match):

Screenshot 2024-01-11 at 11 34 09


We should also underline the "Back" link, it doesn't look it's clickable.

**Comment by lupus2k** [comment]
> This looks better, but we should also use the theme font-family and color.

Hey @crisbusquets, the problem here is the message is inside an iframe, so we can't get any of the theme data, since it's running from jetpack.wordpress.com 😞 
Any other suggestion to make it more generic style message besides the back button underline?

**Review by candy02058912 — DISMISSED** [review]
Tested on:
✅ Chrome
✅ Firefox
✅ Safari (Can only test logged out comments 🤔 )
✅ Edge

**Comment by crisbusquets** [comment]
> > This looks better, but we should also use the theme font-family and color.
> 
> Hey @crisbusquets, the problem here is the message is inside an iframe, so we can't get any of the theme data, since it's running from jetpack.wordpress.com 😞 Any other suggestion to make it more generic style message besides the back button underline?

Oh 😢 
Ok, so what about this approach?

Screenshot 2024-01-12 at 13 28 19

- align text to left (easier to read)
- removed padding-top from h3
- the div that wraps it is now `vertical-align: top;` instead of middle

For the `Comment sent` message, I think we can align it to the center, because the text is shorter. If this exception adds complexity, let's align it to the left.

**Comment by lupus2k** [comment]
Hey @crisbusquets, let me know if it looks good to you now, and I'll merge it! 😉

**Comment by crisbusquets** [comment]
> Hey @crisbusquets, let me know if it looks good to you now, and I'll merge it! 😉

Hi @lupus2k !

There's one minor thing that I don't see: the underline in the "Back" word. Is that possible?

Other than that, the duplicated error message looks better now, thank you :)

**Review by DustyReagan — APPROVED** [review]
Testing well for Atomic in Firefox, Safari, and Chrome.

**Pull request merged**